src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs

```diff
@@ -493,6 +495,24 @@ public static string GetMSBuildFilePath(PackageSpec project, RestoreRequest requ
 
                 props.AddRange(GenerateGroupsWithConditions(buildPropsGroup, isMultiTargeting, frameworkConditions));
 
+                // Create an empty PropertyGroup for package properties
+                var packagePathsPropertyGroup = MSBuildRestoreItemGroup.Create("PropertyGroup", Enumerable.Empty<XElement>(), 1000, isMultiTargeting ? frameworkConditions : Enumerable.Empty<string>());
```

Comment:
Where are you adding PropertyGroup condition for `ExcludeRestorePackageImports`?

Reply:
Those are added on line `573`.